lib/Doctrine/Migrations/Version/DbalExecutor.php

```diff
@@ -324,9 +325,16 @@ private function logResult(Throwable $e, ExecutionResult $result, MigrationPlan
         }
     }
 
-    private function executeResult(MigratorConfiguration $configuration) : void
+    private function executeResult(MigratorConfiguration $configuration, MigrationPlan $plan) : void
```

Comment:
I personally don't like that the check is done here. It is already too late, the migration process already started. Databases as MYSQL are not that good at transactions and altering the schema.

I would suggest to trigger a new event type right before the `if (count($this->sql) !== 0) {` check in the `executeMigration` method.
In that moment the execution of SQL commands did not start yet and is much safer to interrupt it.

Reply:
I would also expect that, when I consider the example use case, if a big table was the reason why an migration was aborted, that the previous executed queries of the same version on the smaller tables are rolled back. Maybe a test can cover that.